test_runner/regress/test_pageserver_generations.py

```diff
@@ -378,3 +378,73 @@ def assert_deletions_submitted(n: int):
 
     assert get_deletion_queue_unexpected_errors(ps_http) == 0
     assert get_deletion_queue_dropped_lsn_updates(ps_http) == 0
+
+
+def test_emergency_mode(neon_env_builder: NeonEnvBuilder, pg_bin: PgBin):
+    neon_env_builder.enable_generations = True
+    neon_env_builder.enable_pageserver_remote_storage(
+        RemoteStorageKind.MOCK_S3,
+    )
+    env = neon_env_builder.init_start(initial_tenant_conf=TENANT_CONF)
+
+    ps_http = env.pageserver.http_client()
+
+    generate_uploads_and_deletions(env)
+
+    env.pageserver.allowed_errors.extend(
+        [
+            # When the pageserver can't reach the control plane, it will complain
+            ".*calling control plane generation validation API failed.*",
+            # Emergency mode is a big deal, we log errors whenever it is used.
+            ".*Emergency mode!.*",
+        ]
+    )
+
+    # Simulate a major incident: the control plane goes offline
+    assert env.attachment_service is not None
+    env.attachment_service.stop()
+
+    # Remember how many validations had happened before the control plane went offline
+    validated = get_deletion_queue_validated(ps_http)
+
+    generate_uploads_and_deletions(env, init=False)
```

Comment:
I already created an issue about this #5492 today.

Reply:
I've opportunistically decreased the size of data written in this test in this PR [a548284](https://github.com/neondatabase/neon/pull/5469/commits/a548284a855e51014923d061a99d531e5fed4059)

Reply:
I think the 200 instead of 20000 should be good enough for now. It seems however that something spiky has appeared in the test suite, like the surprising timeouts I observed in my PR as well..